### PR TITLE
Added SimplePager from SonataDoctrinePhpcrAdminBundle

### PR DESCRIPTION
**Pull request by pulzarraider**
Added SimplePager from SonataDoctrinePhpcrAdminBundle to improve performance of large datasets

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #649, sonata-project/SonataDoctrineORMAdminBundle#297
| License       | MIT
| SonataDoctrineORMAdminBundle PR        | sonata-project/SonataDoctrineORMAdminBundle#415

### CODE REVIEW

File: Admin/Admin.php [hunk 1]
```diff
@@ -897,6 +904,11 @@ public function buildDatagrid()
         foreach ($this->getExtensions() as $extension) {
             $extension->configureDatagridFilters($mapper);
         }
+
+        // set template for SimplePager if it's not already overloaded
+        if ($this->getPagerType() === 'simple' && $this->templates['pager_results'] === 'SonataAdminBundle:Pager:results.html.twig') {
```
Comment (rande) [inline Admin/Admin.php]: This look ugly

File: Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    ...
+
+    class PostAdmin extends Admin
+    {
+        /**
+         * Pager type
+         *
+         * @var string
+         */
+        protected $pagerType = 'simple';
```
Comment (rande) [inline Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst]: It should be also possible to configure this with the main ``sonata_admin`` configuration. The use case is admin from external vendor, it is not possible to create and configure a new class just for altering the pager type.

File: Admin/Admin.php (hunk 1 above)
Comment (pulzarraider) [inline Admin/Admin.php]: Indeed, this is just a small help for user to not need to change the pager template if he change the pager type. Any better ideas are welcome. Inside dependency injection is too sooner for this (the pager type is unknown).

Comment (pulzarraider) [inline Admin/Admin.php]: On second thought I will try to define this using the dependency injection as the prefered way to changing pager type.

File: Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst (hunk 2 above)
Comment (pulzarraider) [inline Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst]: I have added the configuration option `pager_type`.

File: Admin/Admin.php (hunk 1 above)
Comment (pulzarraider) [inline Admin/Admin.php]: Fixed with dependency injection.

File: Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst (hunk 2 above)
Comment (pulzarraider) [inline Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst]: Thank you @rande

**Comment by rande** [comment]
@pulzarraider Is this PR ready ?

**Comment by pulzarraider** [comment]
@rande yes, it's ready.

**Comment by rande** [comment]
@pulzarraider thanks for this work!

**Comment by pulzarraider** [comment]
You're welcome @rande.

**Comment by Renrhaf** [comment]
After updating to master branch : 

```
Fatal error: Call to undefined method Application\Sonata\UserBundle\Admin\UserAdmin::getPagerType() in /var/www/symfony/actresources/vendor/sonata-project/doctrine-orm-admin-bundle/Builder/DatagridBuilder.php on line 136
```

Does it have something to do with this PR ?

**Pull request merged**